### PR TITLE
Implement crypto functionality

### PR DESCRIPTION
**Pull request by tompave-roo**
Proposal for a high level API to encrypt and decrypt ruby Hashes.
Please note that the current implementation has a limit as it's using "plain" RSA encryption instead of an AES cypher, so large data payloads raise these errors:

```
OpenSSL::PKey::RSAError: data too large for key size
```

If the current API looks good, I'll update the implementation and the tests should still pass.

### CODE REVIEW

File: spec/crypto_env_var/cipher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    If a woodchuck could chuck wood.
+    EOS
+  end
+
+  describe "a Cipher initialized with a private key" do
+    subject { described_class.new(private_key) }
+
+    it "can encrypt and decrypt a text" do
+      ciphertext = subject.encrypt(plaintext)
+      expect(ciphertext).to_not eq plaintext
```
Comment (marcinwyszynski) [inline spec/crypto_env_var/cipher_spec.rb]: Nit - a rule I know is one expectation per test case (`it`).

**Comment by tompave-roo** [comment]
> 1) decide on the name of the environment variable that stores the encrypted environment - make it a constant, and use in a convenience wrapper so that using the gem in the app is a one-liner;
> 2) a module to export the hash generated from `encrypt` into `ENV` - to accomplish the above;

Yes, that was the plan.
Copying a Hash into the ENV is quite trivial, but yes, the idea is to have a single method to trigger the whole read-decript-update-ENV process.

File: spec/crypto_env_var/cipher_spec.rb (hunk 1 above)
Comment (tompave-roo) [reply spec/crypto_env_var/cipher_spec.rb]: I know where you're coming from, but I started to appreciate the "it depends" stance.

In this case I'm trying to tell a story and describe the behaviour, and both expectations are required to convey the single concept tested in the example. Does it make sense?

Comment (marcinwyszynski) [reply spec/crypto_env_var/cipher_spec.rb]: Sure. I'm not religious about it.

**Pull request merged**